clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -406,14 +406,13 @@ private FileStatus convertObjectStatsToFileStatus(ObjectStats objectStat) throws
             if (mtime != null) {
                 modificationTime = TimeUnit.SECONDS.toMillis(mtime);
             }
-            Path filePath = new Path(objectStat.getPath()).makeQualified(this.uri, this.workingDirectory);
-
-            URI physicalUri = translateUri(new URI(objectStat.getPhysicalAddress()));
-            Path physicalPath =new Path(physicalUri.toString());
-            FileSystem physicalFS = physicalPath.getFileSystem(conf);
-            long blockSize = physicalFS.getDefaultBlockSize(physicalPath);
-            boolean isdir = objectStat.getPathType() == ObjectStats.PathTypeEnum.COMMON_PREFIX;
-            return new FileStatus(length, isdir, 0, blockSize, modificationTime, filePath);
+            Path filePath = new Path(ObjectLocation.formatPath(repository, ref, objectStat.getPath()));
+            boolean isDir = objectStat.getPathType() == ObjectStats.PathTypeEnum.COMMON_PREFIX;
```

Comment:
I think that it can add clarity if you encapsulate this logic into a private isDir(ObjectStats/pathType) method. it will make life easier for the next person that tries to figure out what we consider as dir, and you can reuse it in line 548.